cmd/lakectl/cmd/dbt.go

```diff
@@ -1,36 +1,90 @@
 package cmd
 
 import (
-	"bufio"
-	"bytes"
 	"context"
-	"encoding/json"
 	"errors"
 	"fmt"
+	"github.com/spf13/cobra"
+	"github.com/treeverse/lakefs/pkg/dbtutil"
+	"github.com/treeverse/lakefs/pkg/metastore"
+	mserrors "github.com/treeverse/lakefs/pkg/metastore/errors"
 	"io/ioutil"
 	"os"
 	"os/exec"
 	"path"
 	"regexp"
-	"strings"
-
-	"github.com/spf13/cobra"
-	"github.com/treeverse/lakefs/pkg/metastore"
-	mserrors "github.com/treeverse/lakefs/pkg/metastore/errors"
 )
 
+type DbtCommandClient interface {
+	// Run performs 'dbt run' over the given schema and selected materializations
+	Run(schema string, selectValues []string) (string, error)
+	// Debug validates dbt connection using dbt debug, return the schema configured by the target environment (configured in the dbt profiles file)
+	Debug() string
+	// Ls performs 'dbt ls' over the given schema and selected materializations
+	Ls(resourceType string, selectStatement []string) []dbtutil.DBTResource
+	// ValidateGenerateSchemaMacro validates that the "generate_schema_name.sql" file was created and used correctly (according to implementation)
+	ValidateGenerateSchemaMacro()
+}
+
+type DbtClient struct {
+	projectDir string
+}
+
 var schemaRegex = regexp.MustCompile(`schema: (.+)`)
 
 const (
-	lakectlIdentifier    = "LAKEFS_SCHEMA"
-	macrosDirName        = "macros"
-	generateSchemaName   = "generate_schema_name.sql"
-	continueOnSchemaFlag = "continue-on-schema-exists"
+	lakectlIdentifier      = "LAKEFS_SCHEMA"
+	macrosDirName          = "macros"
+	generateSchemaName     = "generate_schema_name.sql"
+	continueOnSchemaFlag   = "continue-on-schema-exists"
+	createLakefsBranchFlag = "create-branch"
 )
 
-type model struct {
-	Schema string `json:"schema"`
-	Alias  string `json:"alias"`
+var (
+	noSourceBranchError = errors.New(`given schema has no source lakefs branch associated with it`)
+)
+
+func (d DbtClient) ValidateGenerateSchemaMacro() {
+	err := dbtutil.ValidateGenerateSchemaMacro(d.projectDir, macrosDirName, generateSchemaName, lakectlIdentifier)
+	if err != nil {
+		switch e := err.(type) {
+		case *dbtutil.MissingSchemaIdentifierError:
+			DieFmt(e.Error())
```

Comment:
Why not use `DieErr` in this case?

Reply:
As I understand it, `DieFmt` is used for invalid input. 
In this case the given schema identifier (in the dbt macro) is not the expected `lakectlIdentifier` identifier given here. Although it's not a "classic" input it does seem fit..

Reply:
`DieErr` is the way to die on an error.  `DieFmt` is if you have a message but not an error.  (Yes, one could be implemented in terms of the other.)

In any case, `DieFmt` is dangerous to use like this because the error message may contain formatting chars.  If you must, use `DieFmt("%s", ...)`.